presto-orc/src/main/java/com/facebook/presto/orc/reader/ListStreamReader.java

```diff
@@ -106,14 +106,15 @@ public Block readBlock(Type type)
         // We will use the offsetVector as the buffer to read the length values from lengthStream,
         // and the length values will be converted in-place to an offset vector.
         int[] offsetVector = new int[nextBatchSize + 1];
-        boolean[] nullVector = new boolean[nextBatchSize];
+        boolean[] nullVector = null;
```

Comment:
I expect to see an `Optional.ofNullable` call in this file.  The `ArrayBlock.fromElementBlock` api should be updated to take `Optional<boolean>` for the null block.